nlohmann/json problems

The GHA for the SIRF-SuperBuild fails to compile current STIR `master`, it installs the JSON library. Here are some steps (error is the 3rd quote):

https://github.com/SyneRBI/SIRF/runs/3607921398?check_suite_focus=true#step:6:210
```
-- Installing: /home/runner/install/include/nlohmann/json.hpp
```
During STIR CMake: https://github.com/SyneRBI/SIRF/runs/3607921398?check_suite_focus=true#step:6:8520
```
-- Found nlohmann_json: /home/runner/install/lib/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found suitable version "3.9.1", minimum required is "3.2.0") 
-- nlohmann JSON library found.
```
during STIR compilation
https://github.com/SyneRBI/SIRF/runs/3607921398?check_suite_focus=true#step:6:8862
```
In file included from /home/runner/work/SIRF/SIRF/build/sources/STIR/src/IO/InterfileHeader.cxx:29:
/home/runner/work/SIRF/SIRF/build/sources/STIR/src/include/stir/RadionuclideDB.h:71:10: fatal error: nlohmann/json.hpp: No such file or directory
 #include <nlohmann/json.hpp>
```
but weirdly `RadionuclideDB.cxx` compiles fine, although with warnings https://github.com/SyneRBI/SIRF/runs/3607921398?check_suite_focus=true#step:6:8690
```
/home/runner/work/SIRF/SIRF/build/sources/STIR/src/buildblock/RadionuclideDB.cxx:91:20: warning: comparison of integer expressions of different signedness: ‘int’ and ‘nlohmann::basic_json<>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
     for (int l=0; l<table_json.size(); l++)
                   ~^~~~~~~~~~~~~~~~~~
/home/runner/work/SIRF/SIRF/build/sources/STIR/src/buildblock/RadionuclideDB.cxx:92:24: warning: comparison of integer expressions of different signedness: ‘int’ and ‘nlohmann::basic_json<>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
         for (int c=0; c<table_json.at(0).size(); c++)
                       ~^~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/SIRF/SIRF/build/sources/STIR/src/buildblock/RadionuclideDB.cxx:77:26: warning: control reaches end of non-void function [-Wreturn-type]
     std::string s =this->radionuclide_lookup_table_str;
```

The only thing I can see is that `InterfileHeader` didn't inherit the `include` dependency. A `CMakeLists.txt` problem?

Of course, aside from that, we should remove those warnings.

--- a/src/IO/CMakeLists.txt
+++ b/src/IO/CMakeLists.txt
@@ -93,6 +93,22 @@ if (ITK_FOUND)
   target_link_libraries(IO ITKCommon ${ITK_LIBRARIES})
 endif()
 
+if (HAVE_JSON)
+  # Add the header-only nlohman_json header-only library
+  # Unfortunately, the simple line below exports the dependency while this is really not
+  # necessary.
+  #
+  # target_link_libraries(IO PRIVATE "$<BUILD_INTERFACE:nlohmann_json::nlohmann_json>")
+
+  # So, we currently use an ugly work-around from
+  # https://gitlab.kitware.com/cmake/cmake/-/issues/15415#note_334852
+  # Warning: this will fail once nlohman_json stops being header-only!
+  # In that case, we will need to add it in STIRConfig.cmake.in
+  #
+  get_target_property(TMP nlohmann_json::nlohmann_json INTERFACE_INCLUDE_DIRECTORIES)
+  target_include_directories(IO PRIVATE "${TMP}")
+endif()
+
 # currently needed for ParametricDensity (TODO get rid of this somehow?)
 target_link_libraries(IO modelling_buildblock )
 

--- a/src/buildblock/RadionuclideDB.cxx
+++ b/src/buildblock/RadionuclideDB.cxx
@@ -88,12 +88,14 @@ get_radionuclide_name_from_lookup_table(const std::string& rname) const
         error("The lookup table and the radionuclide database do not have the same number of elements. " 
               "If you added a radionuclide you also need to add the same in the lookup table");
     
-    for (int l=0; l<table_json.size(); l++)
-        for (int c=0; c<table_json.at(0).size(); c++)
+    for (unsigned int l=0; l<table_json.size(); l++)
+        for (unsigned int c=0; c<table_json.at(0).size(); c++)
         {
             if(table_json.at(l).at(c)==rname)
             return table_json.at(l).at(0);
         }
+    /* not found in table, so return as-is */
+    return rname;
 #else
     if (rname.empty())
         return "default";

--- a/src/buildblock/RadionuclideDB.cxx
+++ b/src/buildblock/RadionuclideDB.cxx
@@ -89,7 +89,7 @@ get_radionuclide_name_from_lookup_table(const std::string& rname) const
               "If you added a radionuclide you also need to add the same in the lookup table");
     
     for (unsigned int l=0; l<table_json.size(); l++)
-        for (unsigned int c=0; c<table_json.at(0).size(); c++)
+        for (unsigned int c=0; c<table_json.at(l).size(); c++)
         {
             if(table_json.at(l).at(c)==rname)
             return table_json.at(l).at(0);

